### PR TITLE
Update osm-arc version to 1.2.6-beta-1. 

### PR DESCRIPTION
**Pull request by nshankar13**
- [x] I have ensured that required images are available in MCR:
	1. osm-controller, osm-injector, osm-bootstrap, osm-crds and init images of the corresponding chart version
	2. envoy image of version listed in osm-arc/oss values.yaml
    3. grafana, grafana-image-renderer, prometheus, jaegertracing/all-in-one image of version listed in osm-arc/oss values.yaml
- [x] I have ensured that all the images mentioned in the previous step (aside from grafana-image-renderer) are available for both amd64 and arm64 architectures
- [x] I have updated the Helm chart:
    1. Updated the chart version, app version and dependency version in charts/osm-arc/Chart.yaml    
    2. Made applicable updates to the osm-arc values.yaml if any were made in the upstream OSM chart
    
       
- [x] I have received 3 approvals from maintainers. 

Description: Update chart version to 1.2.6-beta-1. 

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Networking             [ ]
- Metrics                [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [X]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

File: .pipelines/aks-addon-nightly/e2e-aks-addon-jobs.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 jobs:
   - job: test_osm_aks_addon
-    pool: staging-pool
```
Comment (SanyaKochhar) [inline .pipelines/aks-addon-nightly/e2e-aks-addon-jobs.yaml:3]: why is this changing?

**Review by SanyaKochhar — APPROVED** [review]

File: .pipelines/aks-addon-nightly/e2e-aks-addon-jobs.yaml (hunk 1 above)
Comment (nshankar13) [reply .pipelines/aks-addon-nightly/e2e-aks-addon-jobs.yaml:3]: Old staging pools / ADO agents are obsolete - Upstream Publishing is required to use the Mariner agents now.

**Review by ksubrmnn — APPROVED** [review]

File: charts/osm-arc/Chart.yaml [hunk 2]
```diff
@@ -15,14 +15,14 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.2.3
+version: 1.2.6-beta-1
```
[the remark this answers is not in the document]
Comment (nshankar13) [reply charts/osm-arc/Chart.yaml:18]: Actually it looks like it should be 1.2.6.beta*?

[the remark this answers is not in the document]
Comment (nshankar13) [reply charts/osm-arc/Chart.yaml:18]: But maybe better to stick with what we did for the prev Arc release for consistency.

[the remark this answers is not in the document]
Comment (ksubrmnn) [reply charts/osm-arc/Chart.yaml:18]: I've approved for the sake of consistency, but later we can discuss. I'm more familiar with 1.2.6-beta.1

**Review by shalier — APPROVED** [review]

**Pull request merged**